File: ulohy/10-roboti-a-bedny/reseni/main.py
import sys
import math


def read_int() -> int:
    return int(input())


def read_float() -> float:
    return float(input())


def solve_boxes(b: int, r: int, r2: int) -> int:
    return b * r2 // r


def solve_robots(r: int, t: float, t2: float) -> int:
    return math.ceil(r * t / t2)


if __name__ == "__main__":
    b = read_int()
    r = read_int()
    t = read_float()
    c = read_int()

    result = 0

    if c == 1:
        r2 = read_int()
        result = solve_boxes(b, r, r2)
    elif c == 2:
        t2 = read_float()
        result = solve_robots(r, t, t2)

    print(result)

File: ulohy/10-roboti-a-bedny/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let lines = 0;
let b, r, t, c;

rl.on("line", (line) => {
    lines++;
    switch (lines) {
        case 1:
            b = parseInt(line);
            return;

        case 2:
            r = parseInt(line);
            return;

        case 3:
            t = parseFloat(line);
            return;

        case 4:
            c = parseInt(line);
            return;
    }

    let result;

    switch (c) {
        case 1:
            const r2 = parseInt(line);
            result = solveBoxes(b, r, r2);
            break;

        case 2:
            const t2 = parseFloat(line);
            result = solveRobots(r, t, t2);
            break;
    }

    console.log(result);
});

function solveBoxes(b, r, r2) {
    return Math.floor(b * r2 / r);
}

function solveRobots(r, t, t2) {
    return Math.ceil(r * t / t2);
}
